gtime/causality/granger_causality.py

```diff
@@ -0,0 +1,213 @@
+import numpy as np
+import pandas as pd
+
+from scipy import stats
+from sklearn.base import BaseEstimator
+from sklearn.linear_model import LinearRegression
+from sklearn.metrics import mean_squared_error
+
+class GrangerCausality(BaseEstimator):
+    """Class to check for Granger causality between two time series, i.e. 
+    to check if a time series X causes Y: X->Y.
+
+    Parameters
+    ----------
+    target_col : str, 
+        The column to use as the reference, i.e. the time series Y
+    x_col : str
+        The column to test for Granger causality, i.e. the time 
+        series X.
+    max_shift : int
+        The maximal number of shifts to check for Granger causality
+    statistics : str, optional, default: 'ssr_F'
+        The statistical test to perform for Granger causality. Either 'ssr_F'
+        (sum squared residuals with F-test), 'ssr_chi2' (sum squared residuals 
+        with chi square test), 'likelihood_chi2' (likelihood ratio test with 
+        chi square distribution), 'zero_F' (F-test that all lag coefficients of 
+        the time series X are zero) or 'all' (perform all the tests mentioned 
+        above)
+
+    Examples
+    --------
+    >>> from gtime.causality.granger_causality import GrangerCausality
+    >>> import pandas.util.testing as testing
+    >>> data = testing.makeTimeDataFrame(freq="s", nper=1000)
+    >>> GrangerCausality(target_col='A', x_col='B', max_shift=10, statistics='ssr_F').fit(data)
+                        ssr F-test
+    F-value               0.372640
+    p-value               0.958527
+    degrees of freedom  969.000000
+    number of shifts     10.000000
+    """
+    def __init__(self, 
+                 target_col: str, 
+                 x_col: str, 
+                 max_shift: int, 
+                 statistics='ssr_F'):
+        self.target_col = target_col
+        self.x_col = x_col
+        self.max_shift = max_shift
+        self.statistics = statistics
+
+    def fit(self, data: pd.DataFrame):
+        """Create a dataframe with the results of the Granger causality test with the specified
+        statistical test(s).
+
+        Parameters
+        ----------
+        data : pd.DataFrame, shape (n_samples, n_time_series), required
+            The dataframe containing the time series.
+
+        Returns
+        -------
+        result : pd.DataFrame (if only one test is selected) or tuple if statics=='all'
+            A dataframe or a tuple of dataframes for each selected statistical test.
```

Comment:
As a convention, we put a blank line between the last line of the documentation and the """

Reply:
You mean like this:
TEXT

"""

instead of like this:
TEXT
"""

?

Reply:
exactly!

Reply:
I see, then ShiftedLinearCoefficient was probably a bad example for me to look at... because there you don't have the blank line.

Reply:
Oh, then I will fix it. Sorry!